### PR TITLE
fix: Fix docs for ansible-galaxy

### LINKED ISSUE #161: Modules are not documented properly in Ansible Collection
##### SUMMARY
Per https://docs.ansible.com/ansible/latest/dev_guide/developing_collections_documenting.html
Ansible collection documentation must be defined within each module,
```
DOCUMENTATION = r'''
```

https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#module-documenting for more details.

What this project does today is include a docs/ directory (https://docs.ansible.com/ansible/latest/dev_guide/developing_collections_structure.html#collections-doc-dir) and use https://github.com/linode/ansible-specdoc to generate the module documentation within the .py files.

Neither the docs/ documentation nor the representation of that documentation in the .py files appear in the Ansible Galaxy view: 
https://galaxy.ansible.com/ui/repo/published/equinix/cloud/content/module/metal_device/

The one resource that uses the Ansible Galaxy compatible documentation approach is the inventory plugin for Metal Devices: https://galaxy.ansible.com/ui/repo/published/equinix/cloud/content/inventory/metal_device/.

Notice how every other documentation link is empty.

Move the documentation to the modules themselves.

##### ISSUE TYPE
- Documentation Report

##### ANSIBLE VERSION

```paste below

```

### PR DESCRIPTION
**Pull request by t0mk**
We have no docs generated for modules in ansible galaxy. It's because docstrings in module were not valid yaml. This PR fixes specdoc based on which docstrings are generated, and adds a check during docs generation, that ansible-doc can parse module docstrings.

fixes #161

### CODE REVIEW

File: plugins/modules/metal_bgp_session.py [hunk 1]
```diff
@@ -56,13 +56,14 @@
   description: The module object
   returned: always
   sample:
-  - "\n [\n        {\n            \"address_family\": \"ipv4\",\n            \"default_route\"\
```
Comment (ctreatma) [inline plugins/modules/metal_bgp_session.py:59]: I see that you've added a step to test these samples so contributors are directed to fix the formatting of their samples; were the original samples hand-written or were they generated?

Comment (t0mk) [reply plugins/modules/metal_bgp_session.py:59]: @ctreatma The RETURN, EXAMPLES and DOCUMENTAITON are (and always were) generated with [ansible-specdoc](https://github.com/linode/ansible-specdoc) from SPECDOC_META variable in each module. SPECDOC_META is put together from `result_sample`, `specdoc_examples` and `module_spec`. You can see it for example in the metal_gateway module from this branch:
https://github.com/equinix-labs/ansible-collection-equinix/blob/fix_docs_for_galaxy/plugins/modules/metal_gateway.py#L222

The `RETURN` in particular is generated from `result_sample`.

The result_sample is handwritten. We tried to copypaste output samples verbatim from Ansible tests. Ansible output is yaml, so we kept it as string. However, ansible-specdoc did not just copy the result_sample to RETURN, but emitted this mess instead. 

In this PR, I reformatted the strings to python objects, which then ansible-specdoc transforms to valid yaml. This means that in order to write result_sample for new module, we'll have to transcribe example ansible output to Python format.

Comment (ctreatma) [reply plugins/modules/metal_bgp_session.py:59]: OK, makes sense.  Could you make sure that those steps are captured in either the contributing doc or the development doc to help us remember to do them for new modules?

Comment (ctreatma) [reply plugins/modules/metal_bgp_session.py:59]: Added an issue for documenting the documentation process so we can address it in a separate PR

**Review by ctreatma — APPROVED** [review]

**Pull request merged**